GameSessionsManager.h

```diff
@@ -1,19 +1,25 @@
 #ifndef TICTACTOE_GAMESESSION_H
 #define TICTACTOE_GAMESESSION_H
 
-#include "SearchManager.h"
 #include <thread_pool.h>
+#include <GrpcDelegatingWriter.h>
+#include <GrpcDelegatingReader.h>
+#include <Player.h>
+#include "PlayerInfo.h"
 #include <GameSession.h>
 
 class GameSessionsManager {
-	ThreadPool threadpool;
+	ThreadPool game_session_threads;
 
-public:
+	Player create_player(const std::string& player_id, const symbol& symbol_, const std::string& player_address){
+		return Player(player_id, symbol_, new GrpcDelegatingWriter(player_address), new GrpcDelegatingReader(player_address));
+	};
 
-	void start_game_session(SearchManager::PlayerInfo player1, SearchManager::PlayerInfo player2) {
-		threadpool.submit_job([&]() {
-			GameSession session(Player(player1.id, symbol::CROSS, new GrpcDelegatingWriter(player1.adress), new GrpcDelegatingReader(player1.adress)),
-												Player(player2.id, symbol::CIRCLE, new GrpcDelegatingWriter(player2.adress), new GrpcDelegatingReader(player2.adress));
+public:
+	void start_game_session(PlayerInfo player1, PlayerInfo player2) {
+		game_session_threads.submit_job([&]() {
+			GameSession session(create_player(player1.id, symbol::CROSS, player1.address),
```

Comment:
This isn't what I meant with my previous comment about this line. Refactor it to enhance readability. This line contains too much information.